e2etest/bgp_test.go

```diff
@@ -412,25 +453,27 @@ func createServiceWithBackend(cs clientset.Interface, namespace string, policy c
 	return svc, jig
 }
 
-func validateGaugeValue(expectedValue int, metricName string, labels map[string]string, allMetrics []map[string]*dto.MetricFamily) {
-	var err error
-	var value int
+func validateGaugeValue(expectedValue int, metricName string, labels map[string]string, allMetrics []map[string]*dto.MetricFamily) error {
 	found := false
 	for _, m := range allMetrics {
-		value, err = metrics.GaugeForLabels(metricName, labels, m)
+		value, err := metrics.GaugeForLabels(metricName, labels, m)
 		if err != nil {
 			continue
 		}
+		if value != expectedValue {
+			return fmt.Errorf("invalid value %d for %s, expecting %d", value, metricName, expectedValue)
+		}
 		found = true
-		framework.ExpectEqual(value, expectedValue, "invalid value for ", metricName, labels)
 	}
 
 	if !found {
-		framework.Fail(err.Error())
+		return fmt.Errorf("metric %s not found", metricName)
 	}
+	return nil
+
 }
 
-func validateCounterValue(expectedValue int, metricName string, labels map[string]string, allMetrics []map[string]*dto.MetricFamily) {
+func validateCounterValue(expectedMax int, metricName string, labels map[string]string, allMetrics []map[string]*dto.MetricFamily) error {
```

Comment:
why rename from Value to Max , other thank that LGTM

Reply:
because the implementation validates that the value is below a certain threshold, not that it matches the given value.